src/coreclr/nativeaot/Common/src/Internal/Runtime/CompilerHelpers/StartupCodeHelpers.cs

```diff
@@ -170,31 +170,39 @@ internal static void RunModuleInitializers()
 
         private static unsafe void RunInitializers(TypeManagerHandle typeManager, ReadyToRunSectionType section)
         {
-            var initializers = (delegate*<void>*)RuntimeImports.RhGetModuleSection(typeManager, section, out int length);
-            Debug.Assert(length % IntPtr.Size == 0);
-            int count = length / IntPtr.Size;
-            for (int i = 0; i < count; i++)
+            var pInitializers = (byte*)RuntimeImports.RhGetModuleSection(typeManager, section, out int length);
+            Debug.Assert(length % (MethodTable.SupportsRelativePointers ? sizeof(int) : sizeof(nint)) == 0);
+
+            for (byte* pCurrent = pInitializers;
+                pCurrent < (pInitializers + length);
+                pCurrent += MethodTable.SupportsRelativePointers ? sizeof(int) : sizeof(nint))
```

Comment:
Extracting this logic into a property instead of c&p?

Reply:
Hmm, we still need to switch on the size of the element later in the loop, so while it would be fewer characters, whoever is reading the code would now need to scroll to the definition to match it with the inner logic.

The ultimate improvement would be to split the code that reads the data structures from the code that interprets it with a struct iterator. But this is code that we only touch once every few years and mostly just forget it exists, so I didn't spend time to make it pretty.